build.gradle

```diff
@@ -86,9 +89,25 @@ allprojects {
 
 dependencies {
     compileOnly 'com.google.code.findbugs:jsr305:3.0.2'
+    testImplementation project(":")
     testImplementation 'org.spockframework:spock-core:1.3-groovy-2.5'
 }
 
+if (testVersionFromLocalRepository) {
+    group = "new-group-for-root-project"
```

Comment:
This looks like a considerable amount of hack that should be documented. Why do we need this?

Reply:
It seems like you cannot substitute a project dependency for an external dependency with the same GAV coordinates (known limitation of the engine). So I change the GAV coordinates of the root project so the dependency substitution takes effect.

Reply:
I added a comment.